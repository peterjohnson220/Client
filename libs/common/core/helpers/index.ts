export * from './kendo-grid-filter.helper';
