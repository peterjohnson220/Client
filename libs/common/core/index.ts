export * from './pipes';
export * from './helpers';
