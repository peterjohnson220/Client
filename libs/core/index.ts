export * from './helpers';
export * from './pipes';
export * from './common.module';
