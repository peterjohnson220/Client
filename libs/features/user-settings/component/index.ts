export * from './auto-share-modal';
export * from './auto-shared-users';
export * from './shareable-user';
