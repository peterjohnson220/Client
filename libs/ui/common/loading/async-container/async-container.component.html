<div *ngIf="(smartLoadingMask && showLoadingMask) || (!smartLoadingMask && loading) || loadingError"
     class="loading-blanket"
     [style.opacity]="opacityLevel"
     [class.no-opacity]="loadingError || noOpactity">
  <div class="loading-blanket-center">

    <ng-container *ngIf="!loadingError && showSpinner">
      <pf-loading-indicator [spinnerType]="spinnerType" [loadingProgress]="loadingProgress"></pf-loading-indicator>
    </ng-container>

    <ng-container *ngIf="loadingError">

      <div class="d-flex flex-column align-items-center alert alert-danger">
        <pf-error-indicator [message]="loadingErrorMessage || 'Error loading content.'"></pf-error-indicator>

        <ng-container *ngIf="!hideReloadButton">
          <button class="btn btn-danger mt-3"
                  (click)="reload.emit()">{{ reloadText }}</button>
        </ng-container>
      </div>

    </ng-container>

  </div>
</div>

<ng-content></ng-content>
