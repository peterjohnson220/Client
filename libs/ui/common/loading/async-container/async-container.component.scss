@import 'colors';

.loading-blanket{
  background-color: $loading-background-color;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  text-align: center;
}

.loading-blanket-container {
  position: absolute;
  z-index: 100;
  font-size: 20px;
}

.loading-blanket-full-screen {
  position: fixed;
  z-index: 1040;
  font-size: 30px;
}

.loading-blanket-center{
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%,-50%);
}

.no-opacity{
  opacity: 1;
}
