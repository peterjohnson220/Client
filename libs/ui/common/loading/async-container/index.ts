export * from './async-container.component';
