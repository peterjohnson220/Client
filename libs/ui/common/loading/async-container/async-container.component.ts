import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'pf-async-container',
  templateUrl: './async-container.component.html',
  styleUrls: ['./async-container.component.scss']
})
export class AsyncContainerComponent {
  @Input() loading: boolean;
  @Input() loadingError: boolean;
  @Input() loadingErrorMessage: string;
  @Input() reloadText = 'Reload';
  @Input() hideReloadButton: boolean;
  @Input() noOpactity: boolean;
  @Input() spinnerType = 'SVG';

  @Output() reload = new EventEmitter();
}
