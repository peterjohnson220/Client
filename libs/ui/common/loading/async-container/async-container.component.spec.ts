import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { AsyncContainerComponent } from './async-container.component';

describe('Async Container', () => {
  let fixture: ComponentFixture<AsyncContainerComponent>;
  let instance: AsyncContainerComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        AsyncContainerComponent,
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(AsyncContainerComponent);
    instance = fixture.componentInstance;
  });

  it('should show the pf-loading-mask, when it is loading', () => {

    instance.loading = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show the pf-loading-mask, when there is a loading error', () => {

    instance.loadingError = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should not show the pf-loading-mask, when it is not loading and has not had a loading error', () => {

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
