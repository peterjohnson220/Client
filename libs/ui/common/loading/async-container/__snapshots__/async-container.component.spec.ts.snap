// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async Container should not show the .loading-blanket, when it is not loading, and there is no loading error 1`] = `
<pf-async-container
  fullScreenLoadingMask="false"
  loading="false"
  loadingError="false"
  opacityLevel={[Function Number]}
  reload={[Function EventEmitter]}
  reloadText={[Function String]}
  showLoadingMask="false"
  showSpinner={[Function Boolean]}
  smartLoadingMask="false"
  smartLoadingMaskBuffer={[Function Number]}
  smartLoadingMaskWindowMax={[Function Number]}
  smartLoadingMaskWindowMin={[Function Number]}
  spinnerType={[Function String]}
  winRef={[Function Object]}
>
  
</pf-async-container>
`;

exports[`Async Container should show the pf-error-indicator and no reload button, when there is a loading error and hiding reload button 1`] = `
<pf-async-container
  fullScreenLoadingMask="false"
  hideReloadButton={[Function Boolean]}
  loadingError={[Function Boolean]}
  opacityLevel={[Function Number]}
  reload={[Function EventEmitter]}
  reloadText={[Function String]}
  showLoadingMask="false"
  showSpinner={[Function Boolean]}
  smartLoadingMask="false"
  smartLoadingMaskBuffer={[Function Number]}
  smartLoadingMaskWindowMax={[Function Number]}
  smartLoadingMaskWindowMin={[Function Number]}
  spinnerType={[Function String]}
  winRef={[Function Object]}
>
  <div
    class="loading-blanket loading-blanket-container no-opacity"
    ng-reflect-klass="loading-blanket"
    ng-reflect-ng-class="loading-blanket-container"
    style="opacity: 0.8;"
  >
    <div
      class="loading-blanket-center"
    >
      
      
      
      <div
        class="d-flex flex-column align-items-center alert alert-danger"
      >
        <pf-error-indicator />
        
      </div>
    </div>
  </div>
</pf-async-container>
`;

exports[`Async Container should show the pf-error-indicator and reload button, when there is a loading error 1`] = `
<pf-async-container
  fullScreenLoadingMask="false"
  loadingError={[Function Boolean]}
  opacityLevel={[Function Number]}
  reload={[Function EventEmitter]}
  reloadText={[Function String]}
  showLoadingMask="false"
  showSpinner={[Function Boolean]}
  smartLoadingMask="false"
  smartLoadingMaskBuffer={[Function Number]}
  smartLoadingMaskWindowMax={[Function Number]}
  smartLoadingMaskWindowMin={[Function Number]}
  spinnerType={[Function String]}
  winRef={[Function Object]}
>
  <div
    class="loading-blanket loading-blanket-container no-opacity"
    ng-reflect-klass="loading-blanket"
    ng-reflect-ng-class="loading-blanket-container"
    style="opacity: 0.8;"
  >
    <div
      class="loading-blanket-center"
    >
      
      
      
      <div
        class="d-flex flex-column align-items-center alert alert-danger"
      >
        <pf-error-indicator />
        
        
        <button
          class="btn btn-danger mt-3"
        >
          Reload
        </button>
      </div>
    </div>
  </div>
</pf-async-container>
`;

exports[`Async Container should show the pf-loading-indicator, when we are loading 1`] = `
<pf-async-container
  fullScreenLoadingMask="false"
  loading={[Function Boolean]}
  opacityLevel={[Function Number]}
  reload={[Function EventEmitter]}
  reloadText={[Function String]}
  showLoadingMask="false"
  showSpinner={[Function Boolean]}
  smartLoadingMask="false"
  smartLoadingMaskBuffer={[Function Number]}
  smartLoadingMaskWindowMax={[Function Number]}
  smartLoadingMaskWindowMin={[Function Number]}
  spinnerType={[Function String]}
  winRef={[Function Object]}
>
  <div
    class="loading-blanket loading-blanket-container"
    ng-reflect-klass="loading-blanket"
    ng-reflect-ng-class="loading-blanket-container"
    style="opacity: 0.8;"
  >
    <div
      class="loading-blanket-center"
    >
      
      
      <pf-loading-indicator />
      
    </div>
  </div>
</pf-async-container>
`;

exports[`Async Container should transclude 1`] = `
<ng-component>
  <pf-async-container>
    
    <h1>
      Transclusion Works!
    </h1>
  </pf-async-container>
</ng-component>
`;
